### PR TITLE
Packaging fixes

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] release build + tracker issues w/ actions not being plain objects (known issue with sending objects across the ipc bridge)
- [x] update package darwin script to just make the build
- [x] turn back on optimization
- [x] run plain npm commands to retest
- [x] nslog back on

### CODE REVIEW

File: desktop/package.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+const appName = 'Keybase'
+const shouldUseAsar = argv.asar || argv.a || false
+const shouldBuildAll = argv.all || false
+const appVersion = argv.appVersion || '0.0.0'
+const comment = argv.comment || ''
+
+del.sync('build')
+
+fs.copySync('./Icon.png', 'build/desktop/Icon.png')
```
Comment (chrisnojima) [inline desktop/package.js]: For now we explicitly copy assets over, in the future we'll likely want to consolidate where all this stuff goes (like /assets or something)

File: desktop/renderer/index.js [hunk 2]
```diff
@@ -58,7 +58,9 @@ class Keybase extends Component {
     ipcMain.removeAllListeners('stateChange')
     ipcMain.removeAllListeners('subscribeStore')
 
-    ipcMain.on('dispatchAction', (event, action) => {
+    ipcMain.on('dispatchAction', (event, incomingAction) => {
```
Comment (chrisnojima) [inline desktop/renderer/index.js]: this is some tricky shit

File: desktop/renderer/style.css [hunk 3]
```diff
@@ -49,3 +49,12 @@ body {
     margin: 0;
     overflow: hidden;
 }
+
+#app {
+    position: absolute;
```
Comment (chrisnojima) [inline desktop/renderer/style.css]: fixes main tabs from not showing for some reason

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is a branch off of a branch. let's get this approved, merged into the base, then i'll rebase master in

File: desktop/hot-path.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import resolveAssets from './resolve-assets'
+const hot = process.env.HOT === 'true'
+
+export default path => {
+  return (hot ? 'http://localhost:4000/dist/' : resolveAssets('dist') + '/') + path
```
Comment (cjb) [inline desktop/hot-path.js]: Will this be running in prod too?  What if the machine running the Keybase app already has another server listening on port 4000?

**Comment by cjb** [comment]
:+1:%C

File: desktop/hot-path.js (hunk 4 above)
Comment (chrisnojima) [inline desktop/hot-path.js]: We likely shouldn't, it'd be really hard to understand what version of what is running and it'd be slow

File: desktop/renderer/index.js (hunk 2 above)
Comment (MarcoPolo) [inline desktop/renderer/index.js]: wow, how come we didn't run into this sooner?

Comment (chrisnojima) [inline desktop/renderer/index.js]: yeah unclear

**Pull request merged**